### PR TITLE
PYTHON-876: Remove legacy execution parameters in cluster

### PR DESCRIPTION
**Pull request by aboudreault**

### CODE REVIEW

File: cassandra/cluster.py [hunk 1]
```diff
@@ -499,12 +459,10 @@ def default_retry_policy(self, policy):
     connect_to_remote_hosts = True
     """
     If left as :const:`True`, hosts that are considered :attr:`~.HostDistance.REMOTE`
-    by the :attr:`~.Cluster.load_balancing_policy` will have a connection
+    by the :attr:`~.Cluster.profile_manager` will have a connection
     opened to them.  Otherwise, they will not have a connection opened to them.
 
     Note that the default load balancing policy ignores remote hosts by default.
-
-    .. versionadded:: 2.1.0
```
[the remark this answers is not in the document]
Comment (aboudreault) [reply cassandra/cluster.py]: I don't know why this has been removed in the original PR... but adding it back.

File: cassandra/cqlengine/connection.py [hunk 2]
```diff
@@ -79,6 +80,16 @@ def __init__(self, name, hosts, consistency=None,
         self.lazy_connect = lazy_connect
         self.retry_connect = retry_connect
         self.cluster_options = cluster_options if cluster_options else {}
+        ep_kwargs = dict((k, self.cluster_options[k]) for k in ('load_balancing_policy', 'retry_policy') if
```
[the remark this answers is not in the document]
Comment (aboudreault) [reply cassandra/cqlengine/connection.py]: `cluster_options` only act as a proxy. So any invalid keys will raise an exception during Cluster creation. However, I realize that this was mostly for backward compatibility and do not apply for 4.x. The user should send the `execution_profile` if he doesn't want the default one. Will fix.

**Comment by aboudreault** [comment]
Comments addressed. Yes, almost all changes come from the original DSE PR. Hopefully, the OSS -> DSE merge will go well.

File: cassandra/cluster.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mambocab) [reply cassandra/cluster.py]: Maybe it was removed because it wasn't added in DSE 2.1. In any event, +1 to adding it back.

**Comment by mambocab** [comment]
Awesome, feel free to merge when you're happy with the test results.

**Pull request merged**